src/renderer/components/+workloads-cronjobs/cronjob-trigger-dialog.tsx

```diff
@@ -0,0 +1,128 @@
+import "./cronjob-trigger-dialog.scss";
+
+import React, { Component } from "react";
+import { observable } from "mobx";
+import { observer } from "mobx-react";
+import { Trans } from "@lingui/macro";
+import { Dialog, DialogProps } from "../dialog";
+import { Wizard, WizardStep } from "../wizard";
+import { CronJob, cronJobApi, jobApi, Job } from "../../api/endpoints";
+import { Notifications } from "../notifications";
+import { cssNames } from "../../utils";
+import { Input } from "../input";
+import { systemName, maxLength } from "../input/input.validators";
+
+interface Props extends Partial<DialogProps> {
+}
+
+@observer
+export class CronJobTriggerDialog extends Component<Props> {
+  @observable static isOpen = false;
+  @observable static data: CronJob = null;
+
+  @observable jobName = "";
+
+  @observable ready = false;
+
+  static open(cronjob: CronJob) {
+    CronJobTriggerDialog.isOpen = true;
+    CronJobTriggerDialog.data = cronjob;
+  }
+
+  static close() {
+    CronJobTriggerDialog.isOpen = false;
+  }
+
+  get cronjob() {
+    return CronJobTriggerDialog.data;
+  }
+
+  close = () => {
+    CronJobTriggerDialog.close();
+  }
+
+  onOpen = async () => {
+    const { cronjob } = this;
+    this.jobName = cronjob ? cronjob.getName() + "-manual-" + Math.random().toString(36).slice(2, 7) : "";
+    this.jobName = this.jobName.slice(0, 63);
```

Comment:
Why `63`? Please extract to some variable/class-prop and re-use in `maxLength` (if necessary)?

Reply:
What about I add an `rfc1123Label` validator and for that slice just add a comment to the doc to where this restriction comes from ?